react/features/filmstrip/functions.any.js

```diff
@@ -29,19 +31,34 @@ export function updateRemoteParticipants(store: Object, participantId: ?number)
 
     const {
         fakeParticipants,
+        fakeScreenShareParticipants,
         sortedRemoteParticipants,
         sortedRemoteScreenshares,
         speakersList
     } = state['features/base/participants'];
     const remoteParticipants = new Map(sortedRemoteParticipants);
     const screenShares = new Map(sortedRemoteScreenshares);
+    const screenShareParticipants = new Map(fakeScreenShareParticipants);
     const sharedVideos = fakeParticipants ? Array.from(fakeParticipants.keys()) : [];
     const speakers = new Map(speakersList);
 
-    for (const screenshare of screenShares.keys()) {
-        remoteParticipants.delete(screenshare);
-        speakers.delete(screenshare);
+    if (getSourceNameSignalingFeatureFlag(state)) {
+        for (const participant of screenShareParticipants.keys()) {
```

Comment:
I don't think we need `sortedRemoteScreenshares`  in the new flow so I'm using a new `screenShareParticipants` map to not mix the two

Reply:
will take another look into sortedRemoteScreenshares, I think we will need to also have fakeScreenShareParticipants sorted

Reply:
Yes we need to have fakeScreenShareParticipants sorted always.